Plugins should not require removing use_frameworks! from Podfile

Right now the [Google Sign In example app](https://github.com/flutter/google_sign_in/tree/master/example) has removed use_frameworks! from its Podfile.

If this line is added back, the pod install step fails with the following error.

```
[!] The 'Pods-Runner' target has transitive dependencies that include static binaries: (/Users/jackson/git/gsi42/example/ios/Pods/Google/Frameworks/GGLCore.framework and /Users/jackson/git/gsi42/example/ios/Pods/Google/Frameworks/GGLSignIn.framework)
```

See https://github.com/firebase/FirebaseUI-iOS/issues/34 for an example of other devs struggling with this issue.

A related issue is that the FlutterFire plugins are also written assuming use_frameworks! is not in the Podfile. For example, the [Firebase Analytics example app](https://github.com/flutter/firebase_analytics/tree/master/example) fails when use_frameworks! is added back to its Podfile.

```
    <module-includes>:1:1: warning: umbrella header for module 'Flutter' does not include header 'FlutterAsyncMessageListener.h' [-Wincomplete-umbrella]
    #import "Headers/Flutter.h"
    ^
    <module-includes>:1:1: warning: umbrella header for module 'Flutter' does not include header 'FlutterJSONMessageListener.h' [-Wincomplete-umbrella]
    <module-includes>:1:1: warning: umbrella header for module 'Flutter' does not include header 'FlutterMessageListener.h' [-Wincomplete-umbrella]
    3 warnings generated.
    In file included from /Users/jackson/git/firebase_analytics/ios/Classes/FirebaseAnalyticsPlugin.m:7:
    /Users/jackson/git/firebase_analytics/example/ios/Pods/Headers/Private/Firebase/Firebase.h:1:9: fatal error: 'FirebaseAnalytics/FirebaseAnalytics.h' file not found
    #import <FirebaseAnalytics/FirebaseAnalytics.h>
            ^
    3 warnings and 1 error generated.
```

We should try to make our plugin system compatible with use_frameworks!, or if that's not possible for Google Sign In, at least make the Firebase plugins able to build with use_frameworks!.

/cc @mravn-google @jakobr-google @szakarias @mit-mit

--- a/packages/flutter_tools/templates/create/ios.tmpl/Podfile
+++ b/packages/flutter_tools/templates/create/ios.tmpl/Podfile
@@ -6,7 +6,9 @@ if ENV['FLUTTER_FRAMEWORK_DIR'] == nil
 end
 
 target 'Runner' do
-  use_frameworks!
+  # Uncomment this line if you want to use Swift in your app.
+  # Note that some Flutter plugins are not compatible with use_frameworks!
+  # use_frameworks!
 
   # Pods for Runner
 

